tests/components/homewizard/test_config_flow.py

```diff
@@ -355,5 +359,71 @@ def mock_initialize():
             result["flow_id"], {CONF_IP_ADDRESS: "2.2.2.2"}
         )
 
-    assert result["type"] == FlowResultType.ABORT
-    assert result["reason"] == "unknown_error"
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {"base": "network_error"}
+
+
+async def test_reauth_flow(hass, aioclient_mock):
+    """Test reauth flow while API is enabled."""
+
+    mock_entry = MockConfigEntry(
+        domain="homewizard_energy", data={CONF_IP_ADDRESS: "1.2.3.4"}
+    )
+
+    mock_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": config_entries.SOURCE_REAUTH,
+            "entry_id": mock_entry.entry_id,
+        },
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "reauth_confirm"
+
+    device = get_mock_device()
+    with patch(
+        "homeassistant.components.homewizard.config_flow.HomeWizardEnergy",
+        return_value=device,
+    ):
```

Comment:
It's probably good to patch the integration setup function here to avoid setting up the whole integration when finishing the reauth flow.